internal/accounts/create.go

```diff
@@ -134,3 +145,218 @@ func create(
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info(fmt.Sprintf(
+		"%s New account created with address %s and name %s.\n",
+		output.SuccessEmoji(),
+		output.Bold(fmt.Sprintf("0x%s", account.Address().String())),
+		output.Bold(name)),
+	)
+
+	err = saveAccount(loader, state, account, selectedNetwork)
+	if err != nil {
+		return nil, err
+	}
+
+	items = []string{
+		"Here’s a summary of all the actions that were taken",
+		fmt.Sprintf("Added the new account to %s.", output.Bold("flow.json")),
+	}
+	if selectedNetwork != config.DefaultEmulatorNetwork() {
+		items = append(items,
+			fmt.Sprintf("Saved the private key to %s.", privateFile),
+			fmt.Sprintf("Added %s to %s.", privateFile, output.Bold(".gitignore")),
+		)
+	}
+	outputList(log, items, false)
+
+	return onChainAccount, nil
+}
+
+// getAccountCreatedAddressWithPubKey monitors the network for account creation events, if the event
+// contains the public key we are interested in then it extracts the newly created address from the event payload.
+func getAccountCreatedAddressWithPubKey(
+	service *services.Services,
+	pubKey crypto.PublicKey,
+	startHeight uint64,
+) (*flow.Address, error) {
+	lastHeight, err := service.Blocks.GetLatestBlockHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	flowEvents, _ := service.Events.Get([]string{flow.EventAccountKeyAdded}, startHeight, lastHeight, 20, 1) // ignore AN errors since we will retry anyway
+
+	var address *flow.Address
+	for _, block := range flowEvents {
+		events := flowkit.NewEvents(block.Events)
+		address = events.GetAddressForKeyAdded(pubKey)
+		if address != nil {
+			break
+		}
+	}
+
+	if address == nil {
+		if lastHeight-startHeight > 400 { // if something goes wrong don't keep waiting forever to avoid spamming network
+			return nil, fmt.Errorf("failed to get the account address due to time out")
+		}
+
+		time.Sleep(time.Second * 2)
+		address, err = getAccountCreatedAddressWithPubKey(service, pubKey, startHeight)
+		if err != nil {
+			return nil, err
+		}
+
+		return address, nil
+	}
+
+	return address, nil
+}
+
+func saveAccount(
+	loader flowkit.ReaderWriter,
+	state *flowkit.State,
+	account *flowkit.Account,
+	network config.Network,
+) error {
+	state.Accounts().AddOrUpdate(account)
+
+	// If not using emulator, save account private key private file for security.
+	if network != config.DefaultEmulatorNetwork() {
+		privateLocation := fmt.Sprintf("%s.private.json", account.Name())
+		state.SetAccountFileLocation(*account, privateLocation)
+		err := util.AddToGitIgnore(privateLocation, loader)
+		if err != nil {
+			return err
+		}
+	}
+
+	return state.SaveDefault()
+}
+
+// outputList helper for printing lists
+func outputList(log *output.StdoutLogger, items []string, numbered bool) {
+	log.Info(fmt.Sprintf("%s:", items[0]))
+	items = items[1:]
+	for n, item := range items {
+		sep := " -"
+		if numbered {
+			sep = fmt.Sprintf(" %d.", n+1)
+		}
+		log.Info(fmt.Sprintf("%s %s", sep, item))
+	}
+	log.Info("")
+}
```

Comment:
Should this go in the `output` package?

Reply:
It could yeah. Wil move it. Generally, I was thinking about refactoring this interactive UI into a separate file since more commands will introduce this mode and it could be nicer to have an interactive UI component handling that. Not sure yet exactly how, since I want to wait a bit to see the pattern evolving but I have in mind to refactor this down the line, I don't like the command itself becoming bloated with these interactive responses.